webapp/channels/src/components/drafts/scheduled_post_list/scheduled_post_list.tsx

```diff
@@ -1,30 +1,82 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useEffect} from 'react';
-import {useDispatch, useSelector} from 'react-redux';
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
 
-import fetchTeamScheduledPosts from 'mattermost-redux/actions/scheduled_posts';
-import {
-    makeGetScheduledPostsByTeam,
-} from 'mattermost-redux/selectors/entities/scheduled_posts';
+import type {ScheduledPost} from '@mattermost/types/schedule_post';
+import type {UserProfile, UserStatus} from '@mattermost/types/users';
+
+import {hasScheduledPostError} from 'mattermost-redux/selectors/entities/scheduled_posts';
 import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 
+import AlertBanner from 'components/alert_banner';
+import ScheduledPostItem from 'components/drafts/scheduled_post/scheduled_post';
+import NoScheduledPostsIllustration from 'components/drafts/scheduled_post_list/empty_scheduled_post_list_illustration';
+import NoResultsIndicator from 'components/no_results_indicator';
+
 import type {GlobalState} from 'types/store';
 
-export default function ScheduledPostList() {
-    const dispatch = useDispatch();
-    const currentTeamId = useSelector(getCurrentTeamId);
-    const getScheduledPostsByTeam = makeGetScheduledPostsByTeam();
-    const scheduledPosts = useSelector((state: GlobalState) => getScheduledPostsByTeam(state, currentTeamId, true));
+import './style.scss';
+
+type Props = {
+    scheduledPosts: ScheduledPost[];
+    user: UserProfile;
+    displayName: string;
+    status: UserStatus['status'];
+}
 
-    useEffect(() => {
-        dispatch(fetchTeamScheduledPosts(currentTeamId, true));
-    }, [currentTeamId, dispatch]);
+export default function ScheduledPostList({scheduledPosts, user, displayName}: Props) {
+    const {formatMessage} = useIntl();
+
+    const currentTeamId = useSelector(getCurrentTeamId);
+    const scheduledPostsHasError = useSelector((state: GlobalState) => hasScheduledPostError(state, currentTeamId));
 
     return (
-        <div className='harshil'>
-            <h1>{`Scheduled Posts Count: ${scheduledPosts?.length}`}</h1>
+        <div className='ScheduledPostList'>
+            {
+                scheduledPostsHasError &&
+                <AlertBanner
+                    mode='danger'
+                    className='scheduledPostListErrorIndicator'
+                    message={
+                        <FormattedMessage
+                            id='scheduled_post.panel.error_indicator.message'
+                            defaultMessage='One of your scheduled drafts cannot be sent.'
+                        />
+                    }
```

Comment:
As I said before, not a big fan of passing FormattedMessage as a prop, but again, in this particular case it is not really an issue, since AlertBanner doesn't have a `React.memo` and it just renders whatever you pass without extra re-rendering.

I am fine leaving this like this just for consistency.